Lib/fontTools/ttLib/tables/otConverters_test.py

```diff
@@ -28,5 +30,49 @@ def test_write(self):
         self.assertEqual(writer.getData(), deHexStr("0002"))
 
 
+class NameIDTest(unittest.TestCase):
+    converter = otConverters.NameID('NameID', 0, None, None)
+
+    def makeFont(self):
+        nameTable = newTable('name')
+        nameTable.setName(u"Demibold Condensed", 0x123, 3, 0, 0x409)
+        return {"name": nameTable}
+
+    def test_read(self):
+        font = self.makeFont()
+        reader = OTTableReader(deHexStr("0123"))
+        self.assertEqual(self.converter.read(reader, font, {}), 0x123)
+
+    def test_write(self):
+        writer = OTTableWriter(globalState={})
+        self.converter.write(writer, self.makeFont(), {}, 0x123)
+        self.assertEqual(writer.getData(), deHexStr("0123"))
+
+    def test_xmlWrite(self):
+        writer = makeXMLWriter()
+        self.converter.xmlWrite(writer, self.makeFont(), 291,
+                                "FooNameID", [("attr", "val")])
+        xml = writer.file.getvalue().decode("utf-8")
+        self.assertEqual(
+            xml,
+            '<FooNameID attr="val" value="291"/>  <!-- Demibold Condensed -->')
+
+    def test_xmlWrite_missingID(self):
+        logStream = StringIO()
+        handler = logging.StreamHandler(logStream)
+        otConverters.log.addHandler(handler)
+        writer = makeXMLWriter()
+        self.converter.xmlWrite(writer, self.makeFont(), 666,
+                                "Entity", [("attrib", "val")])
+        xml = writer.file.getvalue().decode("utf-8")
+        self.assertEqual(
+            xml,
+            '<Entity attrib="val"'
+            ' value="666"/>  <!-- missing from name table -->')
+        otConverters.log.removeHandler(handler)
+        self.assertEqual(logStream.getvalue().strip(),
```

Comment:
hey Sascha, your way of intercepting log messages works perfectly fine. But just to let you know, in the subset and hmtx test modules we use a `CapturingLogHandler` context manager for a similar purpose (I think it was added by @rsheeter). It makes these sorts of tests more compact and expressive.

E.g.:
https://github.com/fonttools/fonttools/blob/2300b9c16846ec588f191b9cad89e2e403ef2c13/Lib/fontTools/ttLib/tables/_h_m_t_x_test.py#L78
https://github.com/fonttools/fonttools/blob/master/Lib/fontTools/subset/subset_test.py#L161